test/reverse_segfault.asm

```diff
@@ -0,0 +1,10 @@
+;   segfault
```

Comment:
No, it's not. A segfault should not happen at all

If there is still a segfault one of my patches are not yet applied.
The testcase should end with:

reverse_segfault.asm (7): error: Recursion too deep in doop()
Aborting assembly

Please take a look at file "test/test-report.txt". I checked it in. The testcases are only OK if the last line is

"executed 70 tests, 65 OK, 5 failed, result: overall PASS"

Reply:
The 5 failed testcases are only allowed in selftest subdirectory.  Those are for testing the test engine.

Reply:
![image](https://user-images.githubusercontent.com/3540187/91640794-61453500-ea20-11ea-97ec-256e33562f11.png)

This is the segmentation fault that's showing on the Google build server; is this an expected error message?

Reply:
No, it is not.  There should be:

reverse_segfault.asm (7): error: Recursion too deep in doop()
Aborting assembly

Reply:
Sorry: my previous message was the result from the *.lst file. .
In that particular output this line should be

 *                 reverse_segfault     error_code [38] pass

Reply:
The expecded error code is 38 but the test result error code is 139. This false-positive test should be considered as failed.

The problem is here that the test script does not (yet) check the value of the (expected) error code.

Should I improve the test-script further ?

Reply:
I've hardened the test script further in commit da857c1d0cc055487d002aa27029f9d1de2d8f9fda857c1d0cc055487d002aa27029f9d1de2d8f9f

Reply:
Hi Thomas, one of our team members is blocked by the test cases which currently segfault and hang. I assume this is caused by the reverse_segfault.asm test? On the build-server, there is a core-dump, but at least it continues. Apparently on some local dev machines this actually causes a hang, so the segfault seems destructive there.
What does the running the tests show on your machine? How can we make sure it doesn't causes hangs for other developers?

![image](https://user-images.githubusercontent.com/3540187/92446262-da2d5500-f1b5-11ea-8c9a-162dbfef5416.png)

Reply:
> What does the running the tests show on your machine? How can we make sure it doesn't causes hangs for other developers?

The "hanging" is probably due to the recursion that is deepen endlessly. That is the status _before_ my corrections.
When virtual memory exhausted it will segfault. The more RAM you have on your build-machine the longer you wait for the test to end.

**Again, one of my corrections have not been applied there**. It should stop with

```
reverse_segfault.asm (7): error: Recursion too deep in doop()
Aborting assembly
```

in the build output and

* reverse_segfault     error_code [38] pass

in the test report.

Reply:
If it really blocks you, delete all files `test/reverse_segfault.*`

But there will be still that bug inside dasm.

`